src/Microsoft.AspNetCore.Sockets.Client/ITransportFactory.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Logging;
+using System.Net.Http;
+
+namespace Microsoft.AspNetCore.Sockets.Client
+{
+    public interface ITransportFactory
+    {
+        ITransport CreateTransport(TransportType availableServerTransports, ILoggerFactory loggerFactory, HttpClient httpClient);
```

Comment:
This is more of a preference thing. Seems more natural for the logger to go last IMO, but that's just me